Hey @Abelarm, thanks for the PR!

I'm not seeing that error on my end; I'm guessing it's a mypy thing. Though, looking at the build logs, it looks like it's just a linting error that's triggering a travis fail:

```
/home/saiyaman/gitrepos/cloned/mindgraph/mindgraph/graph.py:63:80: E501 line too long (90 > 79 characters)
                    map(functools.partial(node.custom_repr, depth=depth+1), node.threads))
                                                                               ^
/home/saiyaman/gitrepos/cloned/mindgraph/mindgraph/graph.py:66:80: E501 line too long (92 > 79 characters)
                    map(functools.partial(node.custom_repr, depth=depth + 1), node.threads))
                                                                               ^
/home/saiyaman/gitrepos/cloned/mindgraph/mindgraph/graph.py:68:80: E501 line too long (87 > 79 characters)
                return "    "*(depth-1) + "- {}".format(node.name) + ":\n" + '\n'.join(
                                                                               ^
/home/saiyaman/gitrepos/cloned/mindgraph/mindgraph/graph.py:69:80: E501 line too long (92 > 79 characters)
                    map(functools.partial(node.custom_repr, depth=depth + 1), node.threads))
                                                                               ^
```

I cloned and tested your branch on my local machine and it seems to work just fine! Let me know if you want to try and fix it on your end. Otherwise, I'll just change the base branch for this PR, delint it in a separate branch, then merge into master later.

--- a/mindgraph/graph.py
+++ b/mindgraph/graph.py
@@ -54,21 +54,26 @@ class Node(object):
         return self.custom_repr(self, 0)
 
     @staticmethod
-    def custom_repr(node: "Node", depth: int =0) -> str:
+    def custom_repr(node: "Node", depth: int = 0) -> str:
 
         if len(node.threads) > 0:
 
             if depth == 0:
                 return node.name + ":\n" + '\n'.join(
-                    map(functools.partial(node.custom_repr, depth=depth+1), node.threads))
+                    map(functools.partial(node.custom_repr, depth=depth + 1),
+                        node.threads))
             elif depth == 1:
                 return '- ' + node.name + ":\n" + '\n'.join(
-                    map(functools.partial(node.custom_repr, depth=depth + 1), node.threads))
+                    map(functools.partial(node.custom_repr, depth=depth + 1),
+                        node.threads))
             else:
-                return "    "*(depth-1) + "- {}".format(node.name) + ":\n" + '\n'.join(
-                    map(functools.partial(node.custom_repr, depth=depth + 1), node.threads))
+                return "    " * (depth - 1) + "- {}".format(node.name) + \
+                       ":\n" + '\n'.join(map(
+                                         functools.partial(node.custom_repr,
+                                                           depth=depth + 1),
+                                         node.threads))
 
-        return "    "*(depth-1) + "- {}".format(node.name)
+        return "    " * (depth - 1) + "- {}".format(node.name)
 
     def __str__(self) -> str:
         return dump(load(str(self.__repr__())), default_flow_style=False)

--- a/test/test_mindgraph.py
+++ b/test/test_mindgraph.py
@@ -98,6 +98,25 @@ def test_repr(graph):
     ])
 
 
+def test_deep_repr(graph):
+
+    thing2_1 = graph[1][0]
+    assert thing2_1.name == 'another thing within a thing'
+
+    thing2_1.append('super deep thing')
+
+    assert str(graph) == "".join([
+        "learn all the things:\n",
+        "- 1st thing:\n",
+        "  - thing within a thing\n",
+        "  - thing blocking a thing\n",
+        "- 2nd thing:\n"
+        "  - another thing within a thing:\n",
+        "    - super deep thing\n"
+        "  - another thing blocking a thing\n",
+    ])
+
+    thing2_1.pop(0)
 def test_weight_getter_setter():
     node = Node('myNode')
     default_weight = node.weight

--- a/test/test_mindgraph.py
+++ b/test/test_mindgraph.py
@@ -117,6 +117,8 @@ def test_deep_repr(graph):
     ])
 
     thing2_1.pop(0)
+
+
 def test_weight_getter_setter():
     node = Node('myNode')
     default_weight = node.weight

